src/java/org/apache/cassandra/service/reads/DigestResolver.java

```diff
@@ -93,16 +93,14 @@ public PartitionIterator getData()
         {
             // This path can be triggered only if we've got responses from full replicas and they match, but
             // transient replica response still contains data, which needs to be reconciled.
-            DataResolver<E, L> dataResolver = new DataResolver<>(command,
-                                                                 replicaLayout,
-                                                                 (ReadRepair<E, L>) NoopReadRepair.instance,
-                                                                 queryStartNanoTime);
+            DataResolver<E, P> dataResolver
+                    = new DataResolver<>(command, replicaPlan, (ReadRepair<E, P>) NoopReadRepair.instance, queryStartNanoTime);
 
             dataResolver.preprocess(dataResponse);
             // Forward differences to all full nodes
```

Comment:
This comment is wrong now right? We don't forward we just merge.

Reply:
I think this one should be already fixed in CASSANDRA-14698